build-detector.js

```diff
@@ -0,0 +1,30 @@
+const fs = require('fs');
+const path = require('path');
+
+const currentResourceName = GetCurrentResourceName();
+
+// I hate that this file is necessary :(
+
+// This isn't included in release builds or in build process
+// as its only purpose is to detect whether we live in the non-built
+// source code.
+
+on('onResourceStart', async (resName) => {
+  if (resName !== currentResourceName && process.env.NODE_ENV !== 'development') return;
+
+  const currentResourcePath = GetResourcePath(currentResourceName);
+  const targetBuildPath = path.join(currentResourcePath, 'resources', 'dist');
+
+  try {
+    await fs.promises.access(targetBuildPath);
+  } catch {
```

Comment:
Should we also toss a confirmation message saying it contains built files or something letting them know they can read. Easier for debugging.

Reply:
Services are logged under the "info" level when they start.

Reply:
![image](https://user-images.githubusercontent.com/55056068/150372938-e8e62bd3-88fe-4672-8800-3574b0cc8d7c.png)

Yea but they're quite obnoxious and i've never recalled a time where these actually said they didn't start lol.

Reply:
Meh fuck it